### ISSUE TITLE
dependabot: fails to handle creation of PRs with incompatible go module version updates

### ISSUE THREAD

**Issue by brianmcarey** [kubevirt/project-infra#3228]
dependabot should add `+incompatible` to go module versions that do not start with v0 or v1

Without this errors occur when parsing go.mod leading to dependabot failing to create the update PR.

 {"level":"panic","msg":"go.mod:39:2: require github.com/docker/docker: version \"v24.0.7\" invalid: should be v0 or v1, not v24","robot":"dependabot","time":"2024-01-30T13:23:52Z"} 

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-project-infra-dependabot-update/1752320530951704576

**Comment by dhiller** [comment]
While trying to fix this we noticed that there's a `go get ./...` step missing before the make gazelle-update-repos, so gazelle can add the new dependencies that are introduced after the bumps.

(since the docker/docker cve was removed manually the latest build didn't stumble over this any more)

Now we are having problems with the new external dependency `@com_cloudflare_circl` which misses external header files. We found [this bazel gazelle issue](https://github.com/bazelbuild/bazel-gazelle/issues/1393) where folks reported they patched some files to make up for missing stuff.
